packages/libircclient/patch-add-cloudabi

```diff
@@ -0,0 +1,451 @@
+   --enable-openssl        compile with OpenSSL support (no)
++  --enable-examples       compile with examples (yes)
+ 
+ Some influential environment variables:
+   CXX         C++ compiler command
+@@ -2247,6 +2250,14 @@ else
+ fi
+ 
+ 
++# Check whether --enable-examples was given.
++if test "${enable_examples+set}" = set; then :
++  enableval=$enable_examples; with_examples=$enableval
++else
++  with_examples=yes
++fi
++
++
+ # Checks for programs.
+ ac_ext=cpp
+ ac_cpp='$CXXCPP $CPPFLAGS'
+@@ -4535,6 +4546,10 @@ if test "$use_threads" = "yes"; then
+ 	CFLAGS="$CFLAGS -DENABLE_THREADS -D_REENTRANT"
+ fi
+ 
++if test "$with_examples" = "yes"; then
++	BUILD_EXAMPLES=1
++fi
++
+ 
+ # Checking for mingw
+ { $as_echo "$as_me:${as_lineno-$LINENO}: checking For MinGW32" >&5
+@@ -4551,6 +4566,19 @@ $as_echo "yes" >&6; }
+ $as_echo "no" >&6; } ;;
+ esac
+ 
++# Checking for cloudabi
++{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for CloudABI" >&5
++$as_echo_n "checking for CloudABI... " >&6; }
++case "$host" in
++	*-cloudabi*)
++		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
++$as_echo "yes" >&6; }
++		CFLAGS="$CFLAGS -DCLOUDABI"
++		;;
++	*) { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
++$as_echo "no" >&6; } ;;
++esac
++
+ if test "$use_openssl" = "yes"; then
+ 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for CRYPTO_new_ex_data in -lcrypto" >&5
+ $as_echo_n "checking for CRYPTO_new_ex_data in -lcrypto... " >&6; }
+@@ -4662,6 +4690,7 @@ fi
+ 
+ 
+ 
++
+ ac_config_files="$ac_config_files examples/Makefile src/Makefile"
+ 
+ cat >confcache <<\_ACEOF
+--- configure.in
++++ configure.in
+@@ -37,6 +37,12 @@ AC_ARG_ENABLE([openssl],
+             [use_openssl=$enableval],
+             [use_openssl=no])
+ 
++AC_ARG_ENABLE([examples],
++            [AS_HELP_STRING([--enable-examples],
++               [compile with examples (yes)])],
++            [with_examples=$enableval],
++            [with_examples=yes])
++
+ # Checks for programs.
+ AC_PROG_CXX
+ AC_PROG_CC
+@@ -86,6 +92,10 @@ if test "$use_threads" = "yes"; then
+ 	CFLAGS="$CFLAGS -DENABLE_THREADS -D_REENTRANT"
+ fi
+ 
++if test "$with_examples" = "yes"; then
++	BUILD_EXAMPLES=1
++fi
++
+ 
+ # Checking for mingw
+ AC_MSG_CHECKING([For MinGW32])
+@@ -99,6 +109,16 @@ case "$host" in
+ 	*) AC_MSG_RESULT([no]) ;;
+ esac
+ 
++# Checking for cloudabi
++AC_MSG_CHECKING([for CloudABI])
++case "$host" in
++	*-cloudabi*)
++		AC_MSG_RESULT([yes])
++		CFLAGS="$CFLAGS -DCLOUDABI"
++		;;
++	*) AC_MSG_RESULT([no]) ;;
++esac
++
+ if test "$use_openssl" = "yes"; then
+ 	AC_CHECK_LIB(crypto, [CRYPTO_new_ex_data], [], [AC_MSG_ERROR([OpenSSL not found])])
+ 	AC_CHECK_LIB(ssl,    [SSL_library_init], [], [AC_MSG_ERROR([OpenSSL not found])])
+@@ -111,5 +131,6 @@ AC_SUBST(CFLAGS)
+ AC_SUBST(LDFLAGS)
+ AC_SUBST(LIBS)
+ AC_SUBST(PREFIX)
++AC_SUBST(BUILD_EXAMPLES)
+ AC_CONFIG_FILES([examples/Makefile src/Makefile])
+ AC_OUTPUT
+--- examples/Makefile.in
```

Comment:
Well, for sending upstream, you of course want to make this conditional. For the package we're creating here, it would make more sense to come up with the smallest patch needed to solve the job at hand. What about just doing this?

```
--- Makefile
+++ Makefile
@@ -1,6 +1,6 @@
 # $Id: Makefile 84 2009-01-10 01:37:29Z chromerium $
 
-SUBDIRS = src examples
+SUBDIRS = src
 
 all:
        for subdir in $(SUBDIRS); do \
```